themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx

```diff
@@ -202,12 +202,14 @@ class SigningInPage extends React.Component<SigningInPageProps, SigningInPageSta
         });
 
         let updateAIA: AIACommand;
-        if (updateAction) updateAIA = new AIACommand(updateAction, this.props.location.pathname);
+        if (credContainer.updateAction) {
+            updateAIA = new AIACommand(credContainer.updateAction, this.props.location.pathname);
+        }
 
         return (
             <React.Fragment key='userCredentials'> {
                 userCredentials.map(credential => (
-                    <DataListItem aria-labelledby={'credential-list-item-' + credential.userLabel}>
+                    <DataListItem key={'credential-list-item-' + credential.userLabel} aria-labelledby={'credential-list-item-' + credential.userLabel}>
```

Comment:
Should we really use user label as a key? It can contain special character, white spaces, etc.
Or is that ok? AFAIK `key` doesn't appear anywhere in the final HTML, right?

Reply:
No clue if it is safe to use userLabel as a key here or not :)

Just a note that userLabel can be null or empty. And yes, it can contain spaces and national characters with diacritics etc.

Reply:
In line 200 I make sure that userLabel will have a value.  If you don't send me a value for userLabel then it gets set to Msg.localize(credential.type).

The key does not appear on the screen but it does appear in the HTML.  It can be any string.  It just needs to be unique to its siblings in a loop.  This is used by React to identify each item when re-rendering.  It's really not very important for this app, but I include them to keep React from putting a warning in the javascript console.

Come to think of it, I have the ID, so that's really a better choice than userLabel.  I'll fix that soon, but not in this PR.

Reply:
+1 for using ID which is clearly a better choice for stuff like that.

@ssilvert Could you please create a JIRA for this so we don't forget?

Reply:
I went ahead and fixed it then squashed commits.

Reply:
Thanks!